### PR TITLE
Make a working generated angular2.d.ts

### PR DESCRIPTION
**Pull request by alexeagle**
still doesn't work all the way, due to missing symbols

### CODE REVIEW

**Comment by alexeagle** [comment]
Now it is working, your test should pass.

**Comment by alexeagle** [comment]
Here is the .d.ts file at present: https://gist.github.com/alexeagle/c7e0b05927469666d703

**Comment by weswigham** [comment]
:+1:

File: docs/dgeni-package/processors/readTypeScriptModules.js [hunk 1]
```diff
@@ -115,6 +116,22 @@ module.exports = function readTypeScriptModules(tsParser, readFilesProcessor, mo
...
 
   function createExportDoc(name, exportSymbol, moduleDoc, basePath, typeChecker) {
+    exportSymbol.declarations.forEach(function(decl) {
+      var sourceFile = ts.getSourceFileOfNode(decl);
+      if (decl.typeParameters) {
+        name = name + '<' + getText(sourceFile, decl.typeParameters) + '>';
+      }
+      if (decl.heritageClauses) {
+        decl.heritageClauses.forEach(function(heritage) {
+          if (heritage.token == ts.SyntaxKind.ExtendsKeyword) {
```
Comment (mprobst) [inline docs/dgeni-package/processors/readTypeScriptModules.js]: Why?

**Comment by mprobst** [comment]
We'll have to find a proper solution for this ultimately, but I guess this is progress for the time being.

File: docs/dgeni-package/processors/readTypeScriptModules.js (hunk 1 above)
Comment (alexeagle) [inline docs/dgeni-package/processors/readTypeScriptModules.js]: Added implements as well. It does appear in a couple places.

**Comment by alexeagle** [comment]
merged as 7141c15e65583851696c22504b8aa3034cd24715

**Comment by alexeagle** [comment]
Note to anyone following: the generated file doesn't work with definitelytyped, so there is more to do...

**Pull request closed without merging**